src/bsoncxx/lib/bsoncxx/v1/document/view.cpp

```diff
@@ -30,6 +43,167 @@ static_assert(
     is_nothrow_moveable<view::const_iterator>::value,
     "bsoncxx::v1::document::view::const_iterator must be nothrow moveable");
 
+namespace {
+
+constexpr std::uint8_t k_default_view[5] = {5u, 0u, 0u, 0u, 0u};
+
+} // namespace
+
+view::view() : view{k_default_view} {}
+
+view::view(std::uint8_t const* data, std::size_t length) : view{data} {
+    if (length < _empty_length || length < this->size()) {
+        throw v1::exception{v1::document::view::errc::invalid_length};
+    }
+}
+
+view::const_iterator view::cbegin() const {
+    if (!this->operator bool()) {
+        return this->cend();
+    }
```

Comment:
This check (and the one in `find()` below) is [not present in v_noabi](https://github.com/mongodb/mongo-cxx-driver/blob/ad622035ec9c9ab352b42449d5fcbf426e8782e9/src/bsoncxx/lib/bsoncxx/v_noabi/bsoncxx/document/view.cpp#L84). This is a consequence of the new, well-defined "invalid" state of a document view.